homeassistant/components/tuya/binary_sensor.py

```diff
@@ -187,8 +187,9 @@ class TuyaBinarySensorEntityDescription(BinarySensorEntityDescription):
     # https://developer.tuya.com/en/docs/iot/s?id=K9gf48r5zjsy9
     "mc": (
         TuyaBinarySensorEntityDescription(
-            key=DPCode.DOORCONTACT_STATE,
+            key=DPCode.STATUS,
             device_class=BinarySensorDeviceClass.DOOR,
+            on_value="open",
```

Comment:
It seems like the device you are using for testing (if this worked) is not communicating as documented:

![CleanShot 2022-08-26 at 09 34 54](https://user-images.githubusercontent.com/195327/186848637-b83f1516-ac1c-4273-900f-5fcfc39192a4.png)

According to the specifications, it should be `opened`, not `open`.

Reply:
@frenck You are absolutely right. I did notice that as well (and should have called it out, sorry about that).

My device does indeed return `open` and not `opened`, so I would think this is a bug in their documentation. You'll also notice they use the word "Open" in the description column of that table and I think that is also more grammatically correct ("opened" being the past tense of the verb).

In any case, this is what the response from the Tuya API looks like when my door is open (snippet from the device diagnostics in HA):

```
    "status_range": {
      "status": {
        "type": "Enum",
        "value": {
          "range": [
            "closed"
          ]
        }
      }
    },
    "status": {
      "status": "open"
    },
```

and when it is closed:

```
    "status_range": {
      "status": {
        "type": "Enum",
        "value": {
          "range": [
            "closed"
          ]
        }
      }
    },
    "status": {
      "status": "closed"
    },
```